core/src/main/scala/org/broadinstitute/dsde/rawls/snapshot/SnapshotService.scala

```diff
@@ -74,21 +81,42 @@ class SnapshotService(protected val ctx: RawlsRequestContext,
       // prevent disallowed access across workspace or dataset protection boundaries
       snapshotValidator.validateProtectedStatus()
 
+      val wsmPolicyInputs = workspaceAuthDomain.toList match {
+        case Nil => None
+        case authDomainGroups =>
+          Option(new WsmPolicyInputs().inputs(
+            List(
+              new WsmPolicyInput()
+                .namespace("terra")
+                .name("group-constraint")
+                .additionalData(
+                  authDomainGroups.map(groupName => new WsmPolicyPair().key("group").value(groupName)).asJava
+                )
+            ).asJava
+          ))
+      }
+
       try {
         // if there's an existing WSM workspace, make sure its platform is compatible
         // with that of the snapshot's dataset.
         val wsmWorkspace = aggregatedWorkspaceService.fetchAggregatedWorkspace(rawlsWorkspace, ctx)
         snapshotValidator.validateWorkspacePlatformCompatibility(wsmWorkspace.getCloudPlatform)
+
+        // if the existing WSM workspace doesn't have a group-constraint policy, but the Rawls workspace
+        // has an auth domain, backfill the group-constraint policy on the existing WSM workspace
+        if (!wsmWorkspace.policies.exists(policy => policy.namespace.equals("terra") && policy.name.equals("group-constraint")) && wsmPolicyInputs.isDefined) {
```

Comment:
In theory this check isn't strictly necessary. If the workspace has a `group-constraint` policy, then it should be reflected in the auth domain that we fetch earlier. Attempting to update the policy would be a no-op since the groups in the `group-constraint` policy will be the same as the groups in the workspace's auth domain. 

Still, checking here prevents us from an unnecessary WSM call.

Reply:
I wonder if we should error if the `group-constraint` policy is not reflected in the auth domain.

Reply:
We discussed during standup, but putting a summary here as well for posterity.

The idea with policy for the stub Rawls-stage workspaces in WSM is that Rawls will set up `group-constraint` policies at creation time, and then WSM will manage the policies from that point on. 

In the backfill case, we are accounting for the fact that there are existing stub Rawls-stage workspaces that predate Rawls setting any policy on them. 

If an existing Rawls-stage workspace has an auth domain, but it doesn't have a `group-constraint` policy, we interpret that to mean it's an older Rawls-stage workspace and backfill the policy. 

If there is a `group-constraint` policy, then Rawls assumes that WSM knows best and doesn't attempt to compare the policy or otherwise change it.